gateway/compose/production/jupyter/jupyterhub_config.py

```diff
@@ -53,12 +50,36 @@
 # Authenticate users with Auth0
 c.JupyterHub.authenticator_class = "oauthenticator.auth0.Auth0OAuthenticator"
 
-# Enable automatic user creation and allow all users
-c.Authenticator.auto_login = True
-c.Auth0OAuthenticator.allow_all = True
-c.Authenticator.allowed_users = set()
-# Remove explicit user restrictions
-c.Authenticator.admin_users = {"admin"}  # Keep admin user(s)
+# OAuth Authentication Configuration
+c.Authenticator.auto_login = False  # Require explicit login
+c.Auth0OAuthenticator.allow_all = False  # Restrict access to allowed users/groups
+
+# Configure user access control
+allowed_users = os.environ.get("JUPYTERHUB_ALLOWED_USERS", "").split(",")
+allowed_groups = os.environ.get("JUPYTERHUB_ALLOWED_GROUPS", "").split(",")
+
+# Set allowed users if specified
+if allowed_users and allowed_users[0]:
+    c.Authenticator.allowed_users = {
+        user.strip() for user in allowed_users if user.strip()
+    }
+
+# Set allowed groups if specified (Auth0 groups)
+if allowed_groups and allowed_groups[0]:
+    c.Auth0OAuthenticator.allowed_groups = {
+        group.strip() for group in allowed_groups if group.strip()
+    }
+
+# If neither users nor groups are specified, allow all (NOT recommended for production)
+if not (allowed_users and allowed_users[0]) and not (
+    allowed_groups and allowed_groups[0]
+):
+    c.Auth0OAuthenticator.allow_all = True
+    c.Authenticator.allowed_users = set()
+
```

Comment:
this is a production config file, we can remove this block and update the comments in the example .env.